### PR TITLE
Change target file to upload to tmp directory

### PR DESCRIPTION
**Pull request by burgerkingeater**
This is to address issue in https://github.com/linkedin/TonY/pull/203#discussion_r264758879.

### CODE REVIEW

File: tony-core/src/main/java/com/linkedin/tony/TonyClient.java [hunk 1]
```diff
@@ -484,12 +485,23 @@ public void processTonyConfResources(Configuration tonyConf, FileSystem fs) thro
...
+                  new Path(dest.toString()),
+                  new Path(dest.toString()).getName(),
+                  tonyConf,
+                  fs, LocalResourceType.ARCHIVE, jobName);
+            } finally {
+              try {
+                FileUtils.deleteDirectory(tmpDir);
+              } catch (IOException ex) {
+                // ignore the deletion failure and continue
+                LOG.warn("Failed to delete temp directory " + tmpDir);
```
Comment (erwa) [inline tony-core/src/main/java/com/linkedin/tony/TonyClient.java]: Please include `ex`, e.g.: `LOG.warn("Failed ... " + tmpDir, ex)`

File: tony-core/src/main/java/com/linkedin/tony/TonyClient.java [hunk 2]
```diff
@@ -484,12 +485,23 @@ public void processTonyConfResources(Configuration tonyConf, FileSystem fs) thro
...
-                    fs, LocalResourceType.ARCHIVE, jobName);
+            tmpDir.deleteOnExit();
+            try {
+              java.nio.file.Path dest = Paths.get(tmpDir.getAbsolutePath(), file.getName());
+              Utils.zipFolder(Paths.get(resource), dest);
+              Utils.uploadFileAndSetConfResources(appResourcesPath,
+                  new Path(dest.toString()),
+                  new Path(dest.toString()).getName(),
+                  tonyConf,
+                  fs, LocalResourceType.ARCHIVE, jobName);
```
Comment (erwa) [inline tony-core/src/main/java/com/linkedin/tony/TonyClient.java]: I know you didn't name the variable originally, but could you rename `jobName` and `jobNames` to `resourceKey` and `resourceKeys`?

File: tony-core/src/main/java/com/linkedin/tony/TonyClient.java [hunk 3]
```diff
@@ -484,12 +485,23 @@ public void processTonyConfResources(Configuration tonyConf, FileSystem fs) thro
             }
           } else {
```
Comment (erwa) [inline tony-core/src/main/java/com/linkedin/tony/TonyClient.java]: Can you add a comment here `// file is directory`?

Comment (burgerkingeater) [reply tony-core/src/main/java/com/linkedin/tony/TonyClient.java]: updated

File: tony-core/src/main/java/com/linkedin/tony/TonyClient.java (hunk 1 above)
Comment (burgerkingeater) [reply tony-core/src/main/java/com/linkedin/tony/TonyClient.java]: updated, thanks

File: tony-core/src/main/java/com/linkedin/tony/TonyClient.java (hunk 2 above)
Comment (burgerkingeater) [reply tony-core/src/main/java/com/linkedin/tony/TonyClient.java]: updated, thanks

File: tony-core/src/main/java/com/linkedin/tony/TonyClient.java (hunk 3 above)
Comment (burgerkingeater) [reply tony-core/src/main/java/com/linkedin/tony/TonyClient.java]: updated, thanks

**Comment by erwa** [comment]
Looks good. Will merge if tests pass.

**Pull request merged**